great_expectations/dataset/pandas_dataset.py

```diff
@@ -160,8 +160,34 @@ class PandasDataSet(MetaPandasDataSet, pd.DataFrame):
     PandasDataset instantiates the great_expectations Expectations API as a subclass of a pandas.DataFrame.
 
     For the full API reference, please see :func:`DataSet <great_expectations.dataset.base.DataSet>`
+
+    NB
+    1. Subsets of PandaDataSet have ALL the expectations of the original
+       data frame unless the user specifies the discard_failing_expectations=True
+       property on the original data frame.
+    2. Samples of PandaDataSet have ALL the non-failing expectations of the original
+       data frame.
+    3. Concatenations, joins, and merges of PandaDataSets ONLY contain the
+       default_expectations (see :func: `add_default_expectations`)
     """
 
+    discard_failed_expectations = False
```

Comment:
Can you make this an instance variable (move this definition to the constructor) so it's clearer that it's only a property of a particular PandasDataSet (subclass) instance?

Reply:
I thought to make it an instance variable that we need to move it into __init__. Is this correct?

For example,

  def __init__(self, discard_failed_expectations=False, ...):
      self.discard_failed_expectations = discard_failed_expectations
      .
      .
      .

Reply:
Yes, you could put it in init.

I think we might want to come up with a more explicit way to set the default behavior that gets inherited, but for now, if it were in init, your current codepath will then carry over the correct value from other in finalize.

Reply:
Well, that didn't work. It caused nearly all the tests to fail (118 out of 123 according to PyCharm).

Reply:
OK, moving the initialization line after super seemed to fix all the test failures.